src/ngraph/frontend/onnx_import/utils/reduction.hpp

```diff
@@ -0,0 +1,111 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cstddef>  // std::size_t
+#include <cstdint>  // std::int64_t
+#include <iterator> // std::begin, std::end
+#include <memory>   // std::make_shared
+#include <numeric>  // std::iota
+#include <string>
+#include <type_traits> // std::enable_if, std::is_base_of
+#include <vector>
+
+#include "ngraph/axis_set.hpp"
+#include "ngraph/node_vector.hpp"
+#include "ngraph/op/reshape.hpp"
+#include "ngraph/shape.hpp"
+
+#include "core/node.hpp"
+#include "exceptions.hpp"
+#include "utils/common.hpp"
+#include "utils/reshape.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace reduction
+        {
+            namespace detail
+            {
+                inline AxisSet get_reduction_axes(const Node& node)
+                {
+                    auto reduction_axes =
+                        node.get_attribute_value<std::vector<std::size_t>>("axes", {});
+                    if (reduction_axes.empty())
+                    {
+                        reduction_axes = onnx_import::common::get_monotonic_range<std::size_t>(
+                            node.get_ng_inputs().at(0)->get_shape().size());
+                    }
+                    return AxisSet{reduction_axes};
```

Comment:
`AxisSet` is not required unless it is `explicit` then it is required.

Reply:
I think that this code is more readable.